core/tests/protractor_utils/ProfilePage.js

```diff
@@ -70,14 +78,20 @@ var ProfilePage = function() {
   };
 
   this.expectUserToHaveInterestPlaceholder = async function(expectedText) {
+    await waitFor.visibilityOf(interestPlaceholder,
+      'interestPlaceholder label element taking too long to appear');
     expect(await interestPlaceholder.getText()).toMatch(expectedText);
   };
 
   this.expectUserToNotHaveInterestPlaceholder = async function() {
+    await waitFor.visibilityOf(interestPlaceholder,
+      'interestPlaceholder label element taking too long to appear');
     expect(await interestPlaceholder.isPresent()).toBe(false);
   };
 
   this.expectToHaveExplorationCards = async function() {
+     await waitFor.visibilityOf(allExplorationCardElements,
+      'allExplorationCardElements label element taking too long to appear');
```

Comment:
Please wait for a single element, not a list of them. You can use `.first()` for this, but note that this will fail if none are present. It looks like we throw an error in that case anyway though, so it should be okay